Improved error logs and recovery when connection interrupted

Mejiro-McQueen opened the pull request:
AIT currently does not recover from unexpected socket shutdowns.
This forces the user to recover by restarting AIT.

Add logic to recover and report socket shutdowns.

See https://jira.jpl.nasa.gov/browse/SRGDS-113

Commit:
Add Data Flow Graph Plugin

Mejiro-McQueen committed:
Add capability for TCP Manager to publish TCP data into PUB/SUB

Mejiro-McQueen committed:
Improved error logs and recovery when connection interrupted

cSchubes commented on ait/dsn/plugins/TCP.py:
```diff
@@ -238,14 +239,23 @@ def recv_as_client(self):
-                       f"subscription from {self.hostname} {data}")
-                log.debug(msg)
-                return data
+                if not data:
```
I think the docstring for this function might be off, the name is "recv_as_client" but docstrings says "receive data as a server"

cSchubes reviewed (approved):
Looks robust, and the test cases show it works.

Mejiro-McQueen committed:
Integrate VCID Routing

Mejiro-McQueen committed:
Add Graphable class

Graffiti.Graphable can be used to publish and graph Graffiti.Node
Inputs and Outputs are a list: [(edge_name, edge_label)]
